app/controllers/metrics_controller.rb

```diff
@@ -2,18 +2,42 @@ class MetricsController < ApplicationController
 
   # searches in the description of the cases for keywords to order them in categories
   def map_description(description)
-    health = ["hospital", "health", "medical", "psychologist", "pregnant"]
-    legal = ["legal", "interview appointment", "interview prep", "family reunification", "first decision", "second decision"]
-    if legal.any? {|k| description.include? k} and health.any? {|k| description.include? k}
-      return "Legal & Health"
-    elsif legal.any? {|k| description.include? k}
-      return "Legal"
-    elsif health.any? {|k| description.include? k}
-      return "Health"
-    else
-      return "None"
+
+    tags_to_map = [["Health",["hospital", "health", "medical", "psychologist", "pregnant"]],
+                   ["Legal",["legal", "interview appointment", "interview prep", "family reunification", "first decision", "second decision"]],
+                   ["Clothes",["clothes", "freeshop", "free shop"]],
+                   ["Food", ["food"]]]
+    mapped_tags = []
+
+    tags_to_map.each do |tags|
+      if tags[1].any? {|k| description.downcase.include? k}
+        mapped_tags.push(tags[0])
+      end
     end
+    mapped_tags
+
+
   end
+
+  def map_tags(cases)
+    #create map, for every tag empty list to collect cases
+    tags = Case.tag_options.map{|tag| [tag, []]}.to_h
+    cases.each do |c|
+      # if the case has no tags, try to read it out of the description
+      c_tags = c.tags.empty? ? map_description(c.description) : c.tags
+      c_tags.each do |tag|
+        unless tag == ""
+          tags[tag].push(c)
+        end
+      end
+    end
```

Comment:
I tested this with a case that had no tags but "food" in the description and it did not appear on the chart. I was able to get it working by removing the blank tags (not sure why those appear)  earlier:

```
  def map_tags(cases)
    #create map, for every tag empty list to collect cases
    tags = Case.tag_options.map{|tag| [tag, []]}.to_h
    cases.each do |c|
      # if the case has no tags, try to read it out of the description
      c_tags = c.tags.reject(&:blank?).empty? ? map_description(c.description) : c.tags.reject(&:blank?)
      c_tags.each do |tag|
        tags[tag].push(c)
      end
    end

    tags
  end
```